### PR TITLE
Updated std.string.capitalize to use std.uni.asCapitalized internally

### PR DESCRIPTION
**Pull request by JackStouffer**
* on average, this tends to be 2 nanoseconds slower, so completely negligible
* in a separate commit, added missing shortcut to the top menu
* this change revealed a bug in the old implementation that was treated as a feature. `i` was treated as the lower case version of `İ`, when it should be `i\u0307`

### CODE REVIEW

File: std/string.d [hunk 1]
```diff
@@ -2267,7 +2239,7 @@ unittest
         assert(s2 !is s1);
         s1 = to!S("\u0131 \u0130");
         s2 = capitalize(s1);
-        assert(cmp(s2, "\u0049 \u0069") == 0);
+        assert(cmp(s2, "\u0049 i\u0307") == 0);
```
Comment (Hackerpilot) [inline std/string.d]: Do you have a reference I can read to learn why this is correct (and why the old behavior was not)? This merge looks good but I'm not sure how to review this behavior change.

Comment (JackStouffer) [inline std/string.d]: Honestly it was an assumption, coupled with the fact that it's what `asCapitalized ` outputs. It really doesn't make sense to me to think that `İ`'s lower case version is just a regular lower case i.

Comment (quickfur) [inline std/string.d]: Be careful here. There *are* some places in Unicode where the uppercase of the lowercase of something is not equal to the original thing.  I vaguely recall Ali Cehreli mentioning before certain issues involving the upper/lower-casing of the Turkish dotted/undotted i's. Are you absolutely certain this isn't an instance of that?

**Comment by JackStouffer** [comment]
@quickfur according to http://www.fileformat.info/info/unicode/char/0130/index.htm &#304; is a Turkish character and the lowercase version is i. This means that `std.uni.asCapitalized` is outputting the wrong thing, so that needs to be fixed first before this is pulled.

**Comment by JackStouffer** [comment]
@quickfur @Hackerpilot Based upon the evidence presented in https://issues.dlang.org/show_bug.cgi?id=15440#c1, it could go either way. The "simple lowercase" is `\u0069` but the "[proper] lowercase" is `\u0069\u0307`. So it could be argued that std.uni is doing the right thing by not assuming that the letter is Turkish, or it could be argued that std.string is doing the right thing by assuming the letter is Turkish.

**Comment by JackStouffer** [comment]
Python3 outputs `\u0069\u0307` while Python2 outputs `\u0069`. That should say everything.

**Comment by Hackerpilot** [comment]
Because Python 3 is more Unicode-correct?

**Comment by Hackerpilot** [comment]
I read some of the information linked to in issue 15440, and I think that the new behavior of outputting `\u0069\0u0307` makes sense. Unless @quickfur objects, I'm going to merge this.

**Comment by Hackerpilot** [comment]
Auto-merge toggled on

**Comment by quickfur** [comment]
Perhaps somebody should bring this to Ali's attention so that we can have a more informed opinion on how exactly upper-/lowercasing should work with these letters.

**Comment by acehreli** [comment]
This new treatment of iİ is new to me and it makes sense especially hearing that Java 7 went the same way.

Unicode special casing: http://www.unicode.org/Public/UNIDATA/SpecialCasing.txt

And a blog post: http://grepalex.com/2013/02/14/java-7-and-the-dotted--and-dotless-i/

I can't look at this in detail at the moment but we should make sure that dotless-i is handled correctly as well. (I will get back to this in a couple of days.)

**Comment by JackStouffer** [comment]
@Hackerpilot thanks!

**Pull request merged**